### ISSUE TITLE
Azure cannot use machine size Standard_D2s_v3

### ISSUE THREAD

**Issue by ryanlovett** [jupyterhub/zero-to-jupyterhub-k8s#805]
tl;dr: Standard_D2s_v3 bad, Standard_D2_v2 good

The Azure AKS example uses the node type of Standard_D2s_v3, however when I specified this type, az errored (pasted below; formatting mine). Standard_D2_v2 did work and has comparable specs. I think the docs should be changed if someone else can confirm that this node type choice did not work for them in their region. Perhaps it is otherwise fine depending on region, kubernetes version, die roll, etc.

```
az aks create --name ... -g ... \
  --ssh-key-value ... \
  --node-count 3 \
  --node-vm-size Standard_D2s_v3 \
  --kubernetes-version 1.10.5 \
  --output table
Finished service principal creation[##################################]  100.000
Waiting for AAD role to propagate[###                                 ]  10.0000
Waiting for AAD role to propagate[#######                             ]  20.0000
AAD role propagation done[############################################]  100.000
Deployment failed. Correlation ID: ...
Virtual Machine size Standard_D2s_v3 is not in the allowed list
of VM sizes for accelerated networking to be enabled on Network Interface
/subscriptions/../MC_...//providers/Microsoft.Compute/virtualMachines/aks-nodepool....
Allowed sizes:
Standard_D3_v2
Standard_D12_v2
Standard_D3_v2_Promo
Standard_D12_v2_Promo
Standard_DS3_v2
Standard_DS12_v2
Standard_DS13-4_v2
Standard_DS14-4_v2
Standard_DS3_v2_Promo
Standard_DS12_v2_Promo
Standard_DS13-4_v2_Promo
Standard_DS14-4_v2_Promo
Standard_F4
Standard_F4s
Standard_D8_v3
Standard_D8s_v3
Standard_D32-8s_v3
Standard_E8_v3
Standard_E8s_v3
Standard_D3_v2_ABC
Standard_D12_v2_ABC
Standard_F4_ABC
Standard_F8s_v2
Standard_D4_v2
Standard_D13_v2
Standard_D4_v2_Promo
Standard_D13_v2_Promo
Standard_DS4_v2
Standard_DS13_v2
Standard_DS14-8_v2
Standard_DS4_v2_Promo
Standard_DS13_v2_Promo
Standard_DS14-8_v2_Promo
Standard_F8
Standard_F8s
Standard_M64-16ms
Standard_D16_v3
Standard_D16s_v3
Standard_D32-16s_v3
Standard_D64-16s_v3
Standard_E16_v3
Standard_E16s_v3
Standard_E32-16s_v3
Standard_D4_v2_ABC
Standard_D13_v2_ABC
Standard_F8_ABC
Standard_F16s_v2
Standard_D5_v2
Standard_D14_v2
Standard_D5_v2_Promo
Standard_D14_v2_Promo
Standard_DS5_v2
Standard_DS14_v2
Standard_DS5_v2_Promo
Standard_DS14_v2_Promo
Standard_F16
Standard_F16s
Standard_M64-32ms
Standard_M128-32ms
Standard_D32_v3
Standard_D32s_v3
Standard_D64-32s_v3
Standard_E32_v3
Standard_E32s_v3
Standard_E32-8s_v3
Standard_E32-16_v3
Standard_D5_v2_ABC
Standard_D14_v2_ABC
Standard_F16_ABC
Standard_F32s_v2
Standard_D15_v2
Standard_D15_v2_Promo
Standard_D15_v2_Nested
Standard_DS15_v2
Standard_DS15_v2_Promo
Standard_DS15_v2_Nested
Standard_D40_v3
Standard_D40s_v3
Standard_D15_v2_ABC
Standard_M64ms
Standard_M64s
Standard_M128-64ms
Standard_D64_v3
Standard_D64s_v3
Standard_E64_v3
Standard_E64s_v3
Standard_E64-16s_v3
Standard_E64-32s_v3
Standard_F64s_v2
Standard_F72s_v2
Standard_M128s
Standard_M128ms
Standard_L8s_v2
Standard_L16s_v2
Standard_L32s_v2
Standard_L64s_v2
Standard_L96s_v2
SQLGL
SQLGLCore
Standard_D4_v3
Standard_D4s_v3
Standard_D2_v2
Standard_DS2_v2
Standard_E4_v3
Standard_E4s_v3
Standard_F2
Standard_F2s
Standard_F4s_v2
Standard_D11_v2
Standard_DS11_v2
AZAP_Performance_ComputeV17C
Standard_PB6s
Standard_PB12s
Standard_PB24s.
```

The most similar VM type, Standard_D2s_v2, isn't available in the region I'm using, westus2.
```
Operation failed with status: 'Bad Request'. Details: The VM size of AgentPoolProfile:nodepool1 is not allowed in your subscription in location 'westus2'. The available VM sizes are
Standard_A1,
Standard_A1_v2,
Standard_A2,
Standard_A2_v2,
Standard_A2m_v2,
Standard_A3,
Standard_A4,
Standard_A4_v2,
Standard_A4m_v2,
Standard_A5,
Standard_A6,
Standard_A7,
Standard_A8_v2,
Standard_A8m_v2,
Standard_B2ms,
Standard_B2s,
Standard_B4ms,
Standard_B8ms,
Standard_D11_v2,
Standard_D12_v2,
Standard_D13_v2,
Standard_D14_v2,
Standard_D15_v2,
Standard_D16_v3,
Standard_D16s_v3,
Standard_D1_v2,
Standard_D2_v2,
Standard_D2_v3,
Standard_D2s_v3,
Standard_D32_v3,
Standard_D32s_v3,
Standard_D3_v2,
Standard_D4_v2,
Standard_D4_v3,
Standard_D4s_v3,
Standard_D5_v2,
Standard_D64_v3,
Standard_D64s_v3,
Standard_D8_v3,
Standard_D8s_v3,
Standard_DS11_v2,
Standard_DS12_v2,
Standard_DS13-2_v2,
Standard_DS13-4_v2,
Standard_DS13_v2,
Standard_DS14-4_v2,
Standard_DS14-8_v2,
Standard_DS14_v2,
Standard_DS15_v2,
Standard_DS1_v2,
Standard_DS2_v2,
Standard_DS3_v2,
Standard_DS4_v2,
Standard_DS5_v2,
Standard_E16_v3,
Standard_E16s_v3,
Standard_E2_v3,
Standard_E2s_v3,
Standard_E32-16s_v3,
Standard_E32-8s_v3,
Standard_E32_v3,
Standard_E32s_v3,
Standard_E4_v3,
Standard_E4s_v3,
Standard_E64-16s_v3,
Standard_E64-32s_v3,
Standard_E64_v3,
Standard_E64s_v3,
Standard_E8_v3,
Standard_E8s_v3,
Standard_F1,
Standard_F16,
Standard_F16s,
Standard_F16s_v2,
Standard_F1s,
Standard_F2,
Standard_F2s,
Standard_F2s_v2,
Standard_F32s_v2,
Standard_F4,
Standard_F4s,
Standard_F4s_v2,
Standard_F64s_v2,
Standard_F72s_v2,
Standard_F8,
Standard_F8s,
Standard_F8s_v2,
Standard_G1,
Standard_G2,
Standard_G3,
Standard_G4,
Standard_G5,
Standard_GS1,
Standard_GS2,
Standard_GS3,
Standard_GS4,
Standard_GS4-4,
Standard_GS4-8,
Standard_GS5,
Standard_GS5-16,
Standard_GS5-8,
Standard_H16,
Standard_H16m,
Standard_H16mr,
Standard_H16r,
Standard_H8,
Standard_H8m,
Standard_L16s,
Standard_L32s,
Standard_L4s,
Standard_L8s,
Standard_M128-32ms,
Standard_M128-64ms,
Standard_M128ms,
Standard_M128s,
Standard_M64-16ms,
Standard_M64-32ms,
Standard_M64ms,
Standard_M64s,
Standard_NC12,
Standard_NC12s_v2,
Standard_NC24,
Standard_NC24r,
Standard_NC24rs_v2,
Standard_NC24s_v2,
Standard_NC6,
Standard_NC6s_v2,
Standard_ND12s,
Standard_ND24rs,
Standard_ND24s,
Standard_ND6s,
Standard_NV12,
Standard_NV24,
Standard_NV6
```

Format and sort the lists, and run `comm -1 -2 list1.txt list2.txt` to see the intersection of node types.

**Comment by choldgraf** [comment]
huh....is there any reason we can think of why v3 wouldn't work but v2 does work? I am not really sure what the differences are...

**Comment by consideRatio** [comment]
Doh =/

Good guesses - I think either of the following are the reason, as you also suggested if I understood you correctly.
- The guide was made pre K8s 1.10.5, that now requires some new stuff, that the node you requested doesn't support
- The guide was made for a region having that kind of node type available, but your region doesn't

**Comment by ryanlovett** [comment]
Yeah, I think the Azure components just change over time. But I don't know what their infrastructure function looks like, i.e. f(region, kubernetes, date) = ...

Perhaps it would be sufficient to replace "Standard_D2s_v3" with a <NODE_TYPE> variable and suggest that the user find what the node availability is given their region and cluster version requirements. Or add a note saying something like, "this node type might not work for you given x, y, and z."

**Comment by choldgraf** [comment]
yeah I think that works - or just say "your mileage may vary" on Azure in general :-)

**Comment by yuvipanda** [comment]
The 's' usually denotes wether SSD disks can be attached to those nodes or not. Maybe without the s non-ssd disks can't be attached? If the hub dir is a non-ssd disk, and can not be attached to s machines, this would make sense, in as much as anything on Azure makes any kind of sense.

**Comment by consideRatio** [comment]
I think we should close this as it has grown stale and probably outdated. Feel free to reopen or preferably create a new fresh issue if its still a topic to work on!
